administrator/components/com_fields/helpers/fields.php

```diff
@@ -161,6 +161,9 @@ public static function getFields ($context, $item = null, $prepareValue = false)
 				}
 
 				$field->rawvalue = $field->value;
+				
+				//Store item id on field instance for later use			
+				$field->item_id = $item->id;
```

Comment:
I don't see the difference in setting the rawvalue /value and setting the article/item id. Yes the field data are cached but they will always be set per article and the value be specific to the current article that is being rendered.

I have tested this with a test plugin for the editor field type  and a several of other field types and it seems to be working for me.

I would have passed the item into the event directly but in the render method we do not have access to the item when rendering the specific field unless we set the item id in the request which I thought was a bit overkill. Although I am  happy to do that if you think it is better doing it that way.